src/hooks/use-theme-mode.tsx

```diff
@@ -0,0 +1,77 @@
+import { createContext, useContext, useState, useEffect, ReactNode } from 'react';
+import { getIpcApi } from '../lib/get-ipc-api';
+
+export type ThemeMode = 'light' | 'dark' | 'system';
+
+interface ThemeModeContextType {
+	themeMode: ThemeMode;
+	isDarkMode: boolean;
+	setThemeMode: (mode: ThemeMode) => void;
+}
+
+const ThemeModeContext = createContext<ThemeModeContextType | undefined>(undefined);
+
+export const ThemeModeProvider = ({ children }: { children: ReactNode }) => {
+	const [themeMode, setThemeModeState] = useState<ThemeMode>('system');
+	const [isDarkMode, setIsDarkMode] = useState(false);
```

Comment:
It might worth consolidating theme state with language state at some point instead of local state that is spread between multiple Providers/hooks.

Ideally all the preferences are handled in the same low level abstraction/way.

It's fine to be a follow-up I would say but let's not let this kind of thing slip.